File: src/scene/encounter/NullEncounter.java
package scene.encounter;

import model.Notification;
import model.Party;

/**
 * An encounter where nothing happens.
 */
public class NullEncounter extends Encounter {

	public NullEncounter(Party party, int value) {
		super(party, value, EncounterID.NULL);
	}
	
	@Override
	public EncounterNotification doEncounter() {
		value -= 1;
		return makeNotification();
	}

	@Override
	protected EncounterNotification makeNotification() {
		return new EncounterNotification(new Notification("", false), null);
	}

}

File: web/src/scene/encounter/NullEncounter.ts
import { Party } from "../../model/Party";
import { Notification } from "../../model/Notification";
import { Encounter } from "./Encounter";
import { EncounterNotification } from "./EncounterNotification";
import { EncounterID } from "./EncounterID";

export class NullEncounter extends Encounter {
  constructor(party: Party, value: number) {
    super(party, value, EncounterID.NULL);
  }

  doEncounter(): EncounterNotification {
    this.value -= 1;
    return this.makeNotification();
  }

  protected makeNotification(): EncounterNotification {
    return new EncounterNotification(new Notification("", false), null);
  }
}
